superset/views/core.py

```diff
@@ -644,6 +644,18 @@ def explore_json(
             )
             force = request.args.get("force") == "true"
 
+            if datasource_type == "query":
+                return json_errors_response(
+                    errors=[
+                        SupersetError(
+                            message=__(
+                                "This chart type is not supported when using an unsaved query as a chart source. Create a dataset to visualize your data."
```

Comment:
Not sure if we have this pattern in place, but the first sentence here is verbatim with the 1038 error text.  Should we use the value from 1038 variable instead of another string literal with same text?

Reply:
I think they go to different places? I think this is the established pattern in the backend.

Reply:
👍